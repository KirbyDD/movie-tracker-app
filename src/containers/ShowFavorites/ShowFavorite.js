import React, { Component } from 'react';
import  MovieSnippet  from '../MovieSnippet/MovieSnippet'

const ShowFavorites = ({favorites}) => {
  const displayFavorites = favorites[0].map( movie => {
    return <MovieSnippet 
      path={movie.poster_path} 
      key={movie.id} 
      id={movie.id}/>
  })
  return (
    <section>
      <h2>Favorite Movies!</h2>
      {displayFavorites}
    </section>
  )
}

export default ShowFavorites;