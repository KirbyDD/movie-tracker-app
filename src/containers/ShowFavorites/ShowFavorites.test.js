import React from 'react';
import {shallow} from 'enzyme';
import ShowFavorites from './ShowFavorite'

let favorites = [
  {
    id: 1,
    title: 'The Adventure of TronKat',
    poster_path: "/ziEuG1essDuWuC5lpWUaw1uXY2O.jpg",
    release_date: '2019-01-26',
    vote_average: 9.9,
    overview: 'TronKat is the worlds greatest gamer. But when the world needed him most he vanished.',
    movie_id: 1
  },
  {
    id: 2,
    title: 'Booboo the Fool',
    poster_path: "/ziEuG1essDuWuC5lpWUaw1uXY2O.jpg",
    release_date: '2019-11-26',
    vote_average: 4.9,
    overview: 'I aint one of your little friends',
    movie_id: 2
  },
  {
    id: 3,
    title: 'BlueFace Baby',
    poster_path: "/ziEuG1essDuWuC5lpWUaw1uXY2O.jpg",
    release_date: '2019-01-26',
    vote_average: 7.9,
    overview: 'Mopped the floor and hid the wet sign just to catch them slipping.',
    movie_id: 3
  }
]

describe('Show Favorites', () => {
  let wrapper;

  beforeEach(function() {
    wrapper = shallow(
      <ShowFavorites favorites={favorites} />
    )
  })
  it('Should Match Snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})