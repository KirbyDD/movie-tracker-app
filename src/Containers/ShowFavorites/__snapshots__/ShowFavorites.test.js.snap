// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Show Favorites Should Match Snapshot 1`] = `
<section>
  <h2>
    Favorite Movies!
  </h2>
  <MovieSnippet
    id={1}
    key="1"
    path="/ziEuG1essDuWuC5lpWUaw1uXY2O.jpg"
  />
  <MovieSnippet
    id={2}
    key="2"
    path="/ziEuG1essDuWuC5lpWUaw1uXY2O.jpg"
  />
  <MovieSnippet
    id={3}
    key="3"
    path="/ziEuG1essDuWuC5lpWUaw1uXY2O.jpg"
  />
</section>
`;
